### PR TITLE
chore(doc-gen): allow dart cheatsheet to be generated

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -290,16 +294,27 @@ function devGuideExamplesWatch(shredOptions, postShredAction) {
...
+        if (!GENERATE_API_LANGUAGES.indexOf(targetLanguage) !== -1) {
+          // Don't read TypeScript modules if we are not generating API docs - Dart I am looking at you!
+          readTypeScriptModules.$enabled = false;
+        }
         writeFilesProcessor.outputFolder  = targetLanguage + '/latest/api';
+        computePathsProcessor.pathTemplates.push({
+          docTypes: ['cheatsheet-data'],
+          pathTemplate: '../guide/cheatsheet.json',
+          outputPathTemplate: '${path}'
+        });
```
Comment (petebacondarwin) [inline gulpfile.js]: This block should actually be moved to `tools/api-builder/angular.io-package/index.js`

File: public/docs/dart/latest/guide/_data.json [hunk 2]
```diff
@@ -5,6 +5,10 @@
     "title": "Step By Step Guide"
   },
 
+  "cheatsheet": {
+    "title": "Angular Cheat Sheet"
+  },
+
```
Comment (petebacondarwin) [inline public/docs/dart/latest/guide/_data.json]: This adds the cheatsheet to the navigation under the Dart bit of the site - you might not want to do this yet

File: public/docs/js/latest/guide/_data.json [hunk 3]
```diff
@@ -5,6 +5,10 @@
     "title": "Developer Guide"
   },
 
+  "cheatsheet": {
+    "title": "Angular Cheat Sheet"
+  },
+
```
Comment (petebacondarwin) [inline public/docs/js/latest/guide/_data.json]: This adds the cheatsheet to the navigation under the JS bit of the site - you might not want to do this yet

File: public/docs/js/latest/guide/cheatsheet.jade [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.banner
+  p.text-body This cheat sheet is provisional and may change. Angular 2 is currently in Developer Preview.
+
+article(class="l-content-small grid-fluid docs-content")
+  .cheatsheet
+    h2 Angular for JavaScript Cheat Sheet (v2.0.0-alpha.44)
```
Comment (petebacondarwin) [inline public/docs/js/latest/guide/cheatsheet.jade]: We probably need a more automated way of specifying the version number of Angular in these (all three) cheatsheet.jade files

File: public/docs/ts/latest/guide/cheatsheet.jade [hunk 5]
```diff
@@ -16,11 +16,4 @@ article(class="l-content-small grid-fluid docs-content")
...
 
-    table(ng-repeat='section in appCtrl.cheatsheet' ng-cloak)
-      tr
-        th {{section.name}}
-        th(ng-bind-html="appCtrl.getSafeHtml(section.description)")
-      tr(ng-repeat='child in section.items')
-        td
-          code(bold='child.bold',ng-compile) {{child.syntax}}
-        td(ng-bind-html="appCtrl.getSafeHtml(child.description)")
+    ngio-cheatsheet(src='./cheatsheet.json')
```
Comment (petebacondarwin) [inline public/docs/ts/latest/guide/cheatsheet.jade]: Note that we now have a nice Angular directive to display the cheatsheet

**Comment by naomiblack** [comment]
Pete, I get an interpolation error on my test server. See:
screen shot 2015-12-08 at 11 26 37 am

**Comment by petebacondarwin** [comment]
I have fixed the problem - the `<code>` directive does not compile its contents unless the `ng-compile` attribute has a value (e.g. `"true"`)

**Pull request closed without merging**